src/atomWithFormControls.ts

```diff
@@ -0,0 +1,147 @@
+
+  const errorsAtom = atom((get) => {
+    return Object.fromEntries(
+      Object.entries(labeledAtoms).map(([k, v]) => {
+        const val = get(v);
+        // @ts-expect-error atomgroup inference issue
+        if (val.isValid === false) {
+          // @ts-expect-error result of the line above
+          return [k, val.error];
+        }
+        return [k, null];
+      }),
+    );
+  });
+
+  const valueAtom = atom(
+    (get) => {
+      return get(validating);
+    },
+    (get, set, next: ActionableNext) => {
+      switch (next.action) {
+        case 'SET_TOUCHED':
+          return set(touchedState, {
+            ...get(touchedState),
+            [next.key]: Boolean(next.value),
+          });
+        case 'SET_FOCUSED':
+          return set(focusedState, {
+            ...get(focusedState),
+            [next.key]: Boolean(next.value),
+          });
+        case 'SET_VALUE':
+        default: {
+          const actOn = labeledAtoms[next.key];
+          if (!actOn) {
+            return get(validating);
+          }
+          return set(actOn, next.value);
+        }
+      }
+    },
+  );
+
+  const formControlAtom = atom(
+    (get, options) => {
+      const errorVals = get(errorsAtom);
+      const errLen = Object.keys(errorVals).filter((x) => errorVals[x]).length;
+      const validateAtomResult = get(valueAtom);
+      const isValid = validateAtomResult.isValid && errLen === 0;
+
+      // INTERNAL USECASE, AVOID USING IN YOUR OWN LIBS
+      const setter = options.setSelf;
+
+      return {
+        ...validateAtomResult,
+        isValid,
+        errors: errorVals,
+        touched: get(touchedState),
+        focused: get(focusedState),
+        setTouched(key: string, val: boolean) {
+          setter({
+            action: 'SET_TOUCHED',
+            key,
+            value: val,
+          });
+        },
+        setFocused(key: string, val: boolean) {
+          setter({
+            action: 'SET_FOCUSED',
+            key,
+            value: val,
+          });
+        },
+        handleOnChange(key: string) {
+          return (val: any) => {
+            setter({
+              action: 'SET_VALUE',
+              key: key,
+              value: val,
+            });
+          };
+        },
+        handleOnFocus(key: string) {
+          return () =>
+            setter({
+              action: 'SET_FOCUSED',
+              key,
+              value: true,
+            });
+        },
+        handleOnBlur(key: string) {
+          return () => {
+            setter({
+              action: 'SET_TOUCHED',
+              key,
+              value: true,
+            });
+            setter({
+              action: 'SET_FOCUSED',
+              key,
+              value: false,
+            });
+          };
+        },
+      };
+    },
+    (_, set, next: ActionableNext) => set(valueAtom, next),
+  );
+
+  return atom((g) => g(formControlAtom));
```

Comment:
Just to follow the naming convention (it can be minified in the end anyway).
```suggestion
  return atom((get) => get(formControlAtom));
```

Reply:
Still have to polish them up, just sent the pr for approval of implementation